### PR TITLE
Improved serialization tests in Persistence TCK

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
Moved the serialization tests from Persistence.Tests to Persistence TCK

### CODE REVIEW

File: src/contrib/persistence/Akka.Persistence.Sqlite.Tests/Serialization/SqliteJournalSerializationSpec.cs [hunk 1]
```diff
@@ -40,5 +47,15 @@ class = ""Akka.Persistence.Sqlite.Journal.SqliteJournal, Akka.Persistence.Sqlite
                     }
                 }");
         }
+
+        [Fact(Skip = "Sql plugin does not support SerializerWithStringManifest")]
+        public override void Journal_should_serialize_Persistent_with_string_manifest()
```
Comment (alexvaluyskiy) [inline src/contrib/persistence/Akka.Persistence.Sqlite.Tests/Serialization/SqliteJournalSerializationSpec.cs]: Sql plugin does not support SerializerWithStringManifest

File: src/contrib/persistence/Akka.Persistence.Sqlite.Tests/Serialization/SqliteJournalSerializationSpec.cs [hunk 2]
```diff
@@ -40,5 +47,15 @@ class = ""Akka.Persistence.Sqlite.Journal.SqliteJournal, Akka.Persistence.Sqlite
...
+        {
+        }
+
+        [Fact(Skip = "Sql plugin does not support EventAdapter.Manifest")]
+        public override void Journal_should_serialize_Persistent_with_EventAdapter_manifest()
```
Comment (alexvaluyskiy) [inline src/contrib/persistence/Akka.Persistence.Sqlite.Tests/Serialization/SqliteJournalSerializationSpec.cs]: Sql plugin does not support EventAdapter.Manifest

File: src/core/Akka.Persistence.Tests/Serialization/SerializerSpec.cs [hunk 3]
```diff
@@ -97,267 +93,6 @@ public override int GetHashCode()
...
-        }
-
-
-        public override string ToString()
-        {
-            return string.Format("OldPayload({0})", C);
-        }
-    }*/
-
-    public class SnapshotSerializerPersistenceSpec : AkkaSpec
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence.Tests/Serialization/SerializerSpec.cs]: These tests were testing - how custom serializers serialize the persistence messages. I think it is redundant.

File: src/core/Akka.Persistence/Journal/WriteJournal.cs [hunk 4]
```diff
@@ -56,23 +53,20 @@ protected IEnumerable<IPersistentRepresentation> AdaptFromJournal(IPersistentRep
...
-        /// TBD
+        /// INTERNAL API
         /// </summary>
-        /// <param name="representation">TBD</param>
-        /// <returns>TBD</returns>
         protected IPersistentRepresentation AdaptToJournal(IPersistentRepresentation representation)
         {
             var payload = representation.Payload;
             var adapter = _eventAdapters.Get(payload.GetType());
-            representation = representation.WithPayload(adapter.ToJournal(payload));
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence/Journal/WriteJournal.cs]: Do not allocate if we don't need an EventAdapter

File: src/core/Akka.Persistence.Tests/Serialization/SerializerSpec.cs (hunk 3 above)
Comment (Aaronontheweb) [inline src/core/Akka.Persistence.Tests/Serialization/SerializerSpec.cs]: Correct, they are

File: src/core/Akka.Persistence/Journal/WriteJournal.cs (hunk 4 above)
Comment (Aaronontheweb) [inline src/core/Akka.Persistence/Journal/WriteJournal.cs]: Agree

File: src/contrib/persistence/Akka.Persistence.Sqlite.Tests/Serialization/SqliteJournalSerializationSpec.cs (hunk 1 above)
Comment (alexvaluyskiy) [inline src/contrib/persistence/Akka.Persistence.Sqlite.Tests/Serialization/SqliteJournalSerializationSpec.cs]: Yes, it should. `SerializerWithStringManifest` was designed especially for Persistence. But it is not critical and easy to implement.

**Pull request merged**